src/Oscoin/CLI/Command.hs

```diff
@@ -14,25 +14,29 @@ import           Oscoin.CLI.Revision
 import           Oscoin.CLI.Spinner hiding (progress, withSpinner)
 import           Oscoin.Consensus.Mining (mineGenesis)
 import           Oscoin.Consensus.Nakamoto (mineNakamoto)
-import           Oscoin.Crypto.Blockchain.Block (Difficulty)
+import           Oscoin.Crypto.Blockchain.Block (BlockHash, Difficulty)
 import           Oscoin.Crypto.Blockchain.Eval (EvalError(..), buildGenesis)
 import           Oscoin.Crypto.Hash (Hashed)
+import qualified Oscoin.Crypto.Hash as Crypto
 import qualified Oscoin.Crypto.PubKey as Crypto
 import qualified Oscoin.Data.RadicleTx as Rad
 import           Oscoin.Data.Tx (mkTx)
 import           Oscoin.P2P (NodeAddr(..), mkNodeId)
 import           Oscoin.Time (Timestamp)
 
 
+import           Codec.Serialise (Serialise)
+import qualified Crypto.Data.Auth.Tree.Internal as AuthTree
```

Comment:
A little strange that we need to import the `Internal` module here.

Reply:
It was alas an oversight of mine when patching up `avl-tree`, because the `MerkleHash` typeclass indeed lives in the `Internal` module, but it should have probably have lived in something like `.Class`. Happy to fire up a separate PR for that (if you think this can wait up until after merging this guy, but I am happy either way).

Reply:
Merged the other PR.

Reply:
Happy to wait until this is merged :+1: